HPCC-20169 Spark configuration and runtime

Michael-Gardner opened the pull request:
<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

Michael-Gardner commented:
@xwang2713 Please review the changes made to your packaging code.

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 14498b24943e96b627dddd4b48666e55e8aa3e01
Build: success
Install hpccsystems-platform-community_7.0.0-trunk0.el7.x86_64.rpm
HPCC Start: Fail
HPCC start:
Starting mydafilesrv ...     [   OK    ]  
Starting mydali ...          [   OK    ]  
Starting mydfuserver ...     [   OK    ]  
Starting myeclagent ...      [   OK    ]  
Starting myeclccserver ...   [   OK    ]  
Starting myeclscheduler ...  [   OK    ]  
Starting myesp ...           [ TIMEOUT ] 
Starting myroxie ...         [   OK    ]  
Starting mysasha ...         [   OK    ]  
Starting mythor ...          [   OK    ]  
 
HPCC status:
mydafilesrv     ( pid    12429 ) is running ...
mydali          ( pid    12753 ) is running ...
mydfuserver     ( pid    13096 ) is running ...
myeclagent      ( pid    13447 ) is running ...
myeclccserver   ( pid    13782 ) is running ...
myeclscheduler  ( pid    14130 ) is running ...
myesp           is stopped,
myroxie         ( pid    15694 ) is running ...
mysasha         ( pid    16103 ) is running ...
mythor          ( pid    16684 ) is running with 1 slave process(es) ...
myesp           is stopped,
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   33 sec (00:00:33) |  173 sec (00:02:53) |   61 sec (00:01:01) |    9 sec (00:00:09) |  151 sec (00:02:31) |    0 sec (00:00:00) |    0 sec (00:00:00) |  427 sec (00:07:07) |

xwang2713 reviewed (commented):
Looks good.

richardkchapman commented on initfiles/bash/etc/systemd/system/CMakeLists.txt:
```diff
@@ -37,18 +37,24 @@ set(dfuserver "dfuserver" "--daemon %i" "dafilesrv.service")
 set(eclagent "agentexec" "--daemon %i" "dafilesrv.service")
 set(eclccserver "eclccserver" "--daemon %i" "dafilesrv.service")
 set(eclscheduler "eclscheduler" "--daemon %i" "dafilesrv.service")
-set(esp "esp" "--daemon %i" "dafilesrv.service")
```
Why was this line deleted? Seems unrelated (and may explain the smoketest failure?)

Michael-Gardner replied:
This line was deleted because there is now an esp@instance.service file, and we don't generate it from the component@instance.service template.

richardkchapman commented on initfiles/bash/etc/systemd/system/component@instance.service.in:
```diff
@@ -7,7 +7,7 @@ Requires=hpcc-conf@%i.service
 User=@RUNTIME_USER@
 Group=@RUNTIME_GROUP@
-PermissionsStartOnly=true
+PermissionsStartOnly=false
```
Is this related? I've seen this change in a different PR...

richardkchapman commented on initfiles/bash/etc/systemd/system/esp@instance.service.in:
```diff
@@ -0,0 +1,27 @@
+[Unit]
```
Is this related? Not sure how it is connected with Spark

Michael-Gardner replied:
This file is needed because we now start spark along with the esp.  That's why we broke it out into it's own separate file instead of having it get built by a template.

richardkchapman commented on initfiles/bash/etc/systemd/system/thorslave@instance.service.in:
```diff
@@ -6,10 +6,12 @@ After=network.target remote-fs.target @dependencies@
 User=@RUNTIME_USER@
 Group=@RUNTIME_GROUP@
-PermissionsStartOnly=true
+PermissionsStartOnly=false
```
is this related?

Michael-Gardner replied:
This one should be in here.  The one in another PR is regarding the thor@instance.service.in file I believe, but I'll double check.

richardkchapman commented:
@Michael-Gardner Some concerns about changes that seem unrelated, and the smoketest failure

xwang2713 commented:
@Michael-Gardner if "-DUSE_SPARK" turn off which I believe the smoke test does do we skip Spark configuration and startup when HPCC started?

Michael-Gardner commented:
Looking into the smoke test failure

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 830a5944a30cb381fdbd647869503f5562f964e6
Build: success
Install hpccsystems-platform-community_7.0.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |804 |804 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   15 sec (00:00:15) |  174 sec (00:02:54) |   61 sec (00:01:01) |    7 sec (00:00:07) |   28 sec (00:00:28) | 1304 sec (00:21:44) |   18 sec (00:00:18) | 1607 sec (00:26:47) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: a09ee75503b140bd96652cf81e1b0e5038151ec6
Build: success
Install hpccsystems-platform-community_7.0.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |804 |804 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   20 sec (00:00:20) |  174 sec (00:02:54) |   61 sec (00:01:01) |    9 sec (00:00:09) |   30 sec (00:00:30) | 1289 sec (00:21:29) |   19 sec (00:00:19) | 1602 sec (00:26:42) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: b38a268b25901e6555f9b6d9d5a3448d498439fd
Build: success
Install hpccsystems-platform-community_7.0.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |804 |804 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   11 sec (00:00:11) |  173 sec (00:02:53) |   62 sec (00:01:02) |    9 sec (00:00:09) |   30 sec (00:00:30) | 1310 sec (00:21:50) |   20 sec (00:00:20) | 1615 sec (00:26:55) |

rpastrana commented on spark/spark-defaults.conf.in:
```diff
@@ -0,0 +1,35 @@
+# spark.executor.extraJavaOptions	-Xint
+
+# Path for native libraries
+spark.driver.extraLibraryPath @LIB_PATH@
```
where is LIB_PATH set? what is it set to?

Michael-Gardner replied:
generally it will be set to /opt/HPCCSystems/lib

rpastrana commented on spark/spark-env.sh.in:
```diff
@@ -0,0 +1,79 @@
+# - MKL_NUM_THREADS=1        Disable multi-threading of Intel MKL
+# - OPENBLAS_NUM_THREADS=1   Disable multi-threading of OpenBLAS
+
+ALL_IPS=$(/opt/HPCCSystems/sbin/configgen -env /etc/HPCCSystems/environment.xml -listall2 | awk -F , '{print $3}' | sort | uniq)
```
can these paths be generated based on the CMAKE vars we set up for install location, etc?

rpastrana reviewed (approved):
@Michael-Gardner just a couple of comments/questions. 
Also, as Richard pointed out, there seems to be some overlap w/ another commit, do we need a rebase?

rpastrana reviewed (changes requested):
@Michael-Gardner as is, the startup of ESP/Thor will be affected by the startup of spark, we should consider a independent hpcc-spark startup component.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: e69ccce1c772eba4b85f642754b517ddf58ca9dd
Build: success
Install hpccsystems-platform-community_7.0.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   25 sec (00:00:25) |  176 sec (00:02:56) |   63 sec (00:01:03) |    7 sec (00:00:07) |   28 sec (00:00:28) | 1309 sec (00:21:49) |   17 sec (00:00:17) | 1625 sec (00:27:05) |

Commit:
HPCC-20169 Spark runtime and configuration for HPCC Platform

HPCC-20169 Multiple changes to systemd, and configmgr component addition
HPCC-20169 init_spark changes
HPCC-20169 Added init_sparkthor coding
HPCC-20169 systemd changes from spark -> sparkthor
HPCC-20169 Conditionally add Spark information if using spark
HPCC-20169 Default sparkthor in environment.xml, change script names to sparkthor, switch to using daliadmin, strip unneeded variables from sparkThor.xsd and xsl transformations, add daliServers to sparkThor.xsd and xsl
HPCC-20169 Added wait to init_sparkthor, systemd spark conf change
HPCC-20169 Removed daliServers from sparkThor.xsd and fetch from thorNode during transformation
HPCC-20169 Fixes for remote fetch of spark-env.sh

Signed-off-by: Michael Gardner <michael.gardner@lexisnexisrisk.com>

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 61317ba3c246bedc29ac68099ceecab444554c6e
Number of Error(s): 2
CMake Error: File /mnt/disk1/home/vamosax/smoketest/PR-11526/HPCC-Platform/initfiles/bash/etc/systemd/system/spark-hpcc@instance.service.in does not exist.

[Build error:](https://gist.github.com/1379b1f401fc84d8f14213f2341a894e)
Build log [Build log](https://gist.github.com/HPCCSmoketest/1379b1f401fc84d8f14213f2341a894e#file-result-18-08-14-12-46-03-log)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: a258320c11ec2cbc4e5a9355ebe779d1fe5c8a44
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   31 sec (00:00:31) |  173 sec (00:02:53) |   95 sec (00:01:35) |   20 sec (00:00:20) |   42 sec (00:00:42) | 1340 sec (00:22:20) |   18 sec (00:00:18) | 1719 sec (00:28:39) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 09c34586f1840ba93a007e76d90a6575f0f3f732
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   14 sec (00:00:14) |  173 sec (00:02:53) |   93 sec (00:01:33) |   22 sec (00:00:22) |   44 sec (00:00:44) | 1333 sec (00:22:13) |   18 sec (00:00:18) | 1697 sec (00:28:17) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 871f2a28fea78511235e1cdd9a453aabc62cc325
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   11 sec (00:00:11) |  172 sec (00:02:52) |   94 sec (00:01:34) |   19 sec (00:00:19) |   41 sec (00:00:41) | 1320 sec (00:22:00) |   18 sec (00:00:18) | 1675 sec (00:27:55) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 60e439d7f02edf9522b72c3881ecaee155bc253f
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   22 sec (00:00:22) |  170 sec (00:02:50) |   93 sec (00:01:33) |   19 sec (00:00:19) |   42 sec (00:00:42) | 1344 sec (00:22:24) |   18 sec (00:00:18) | 1708 sec (00:28:28) |

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 3396b7d40176580d0a50c2b78618318e6d1122d4
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |804 |1 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


[Errors:](https://gist.github.com/4ce655e586d9d397cb326ca6b6158307)
- test
  - hthor
    - 57\. [badindex (newIndexReadMapping=false)](https://gist.github.com/HPCCSmoketest/4ce655e586d9d397cb326ca6b6158307#file-badindex-newindexreadmapping-false-w20180821-134230-4-log), 

HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   13 sec (00:00:13) |  170 sec (00:02:50) |   93 sec (00:01:33) |   21 sec (00:00:21) |   43 sec (00:00:43) | 1346 sec (00:22:26) |   20 sec (00:00:20) | 1706 sec (00:28:26) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 7bef3304dd6dfb1cb975b181338e1070efede909
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   28 sec (00:00:28) |  171 sec (00:02:51) |   93 sec (00:01:33) |   20 sec (00:00:20) |   42 sec (00:00:42) | 1386 sec (00:23:06) |   16 sec (00:00:16) | 1756 sec (00:29:16) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 0736a6d36fe2b981d0c072495b44fc0156303dbd
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   10 sec (00:00:10) |  174 sec (00:02:54) |   93 sec (00:01:33) |   19 sec (00:00:19) |   41 sec (00:00:41) | 1347 sec (00:22:27) |   19 sec (00:00:19) | 1703 sec (00:28:23) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 4071e6c0005ff3fe258116c2cf37ef29722cd8ae
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   10 sec (00:00:10) |  172 sec (00:02:52) |   93 sec (00:01:33) |   21 sec (00:00:21) |   43 sec (00:00:43) | 1501 sec (00:25:01) |   18 sec (00:00:18) | 1858 sec (00:30:58) |

JamesDeFabia commented:
@g-pan Please review the docID tags

g-pan commented on initfiles/componentfiles/configxml/spark.xsd:
```diff
@@ -20,88 +20,204 @@
+            <!--DOC-Autobuild-code-->
+            <xs:annotation>
+                <xs:appinfo>
+                    <docid>Th.t1</docid>
```
please change to <docid>Sprk.t1</docid>

g-pan reviewed (commented):
please change the <docid> value in the spark.xsd file, if possible

rpastrana commented on initfiles/bash/etc/systemd/system/CMakeLists.txt:
```diff
@@ -29,7 +29,9 @@ set(componentList
     thorslave
-    backupnode)
+    backupnode
+    spark
```
why do we have two spark related components listed here, and should they be renamed?

rpastrana commented on initfiles/bin/init_sparkthor.in:
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+################################################################################
+#    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems®.
```
update date.

rpastrana commented on initfiles/componentfiles/configxml/cgencomplist_win.xml:
```diff
@@ -70,6 +70,9 @@
     <File name="backupnode.xsl" method="xslt" destName="backupnode.xml"/>
   </Component>
+  <Component name="sparkthor" processName="SparkThor" schema="sparkThor.xsd">
+    <File name="sparkThor.xsl" method="xslt" destName="spark-env.conf"/>
```
this entry differs between cgencomplist_win.xml and cgencomplist_linux.xml by the desName:
"spark-env.conf" vs "spark-env.sh" is this intentional?

rpastrana commented on initfiles/componentfiles/configxml/environment.xsd:
```diff
@@ -795,7 +795,7 @@
         </xs:keyref>
         <xs:key name="processKey">
-            <xs:selector xpath=".//EclServerProcess|.//EclAgentProcess|.//SybaseProcess|.//SpareProcess|.//HoleCluster|.//ThorCluster|.//DaliServer|.//BackupNodeProcess"/>
+		<xs:selector xpath=".//EclServerProcess|.//EclAgentProcess|.//SybaseProcess|.//SpareProcess|.//HoleCluster|.//ThorCluster|.//DaliServer|.//BackupNodeProcess|.//SparkThorProcess"/>
```
minor, but indet seems off

rpastrana commented on initfiles/componentfiles/configxml/sparkThor.xsd:
```diff
@@ -0,0 +1,232 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+################################################################################
+#    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems®.
```
date

rpastrana commented on initfiles/componentfiles/configxml/sparkThor.xsd:
```diff
@@ -0,0 +1,232 @@
+################################################################################
+-->
+
+<!-- edited with XMLSPY v5 rel. 4 U (http://www.xmlspy.com) by Rajeev Moondhra (Seisint, Inc.) -->
```
lets remove this copy/paste remnant

rpastrana commented on initfiles/componentfiles/configxml/sparkThor.xsd:
```diff
@@ -0,0 +1,232 @@
+    <xs:include schemaLocation="environment.xsd"/>
+    <xs:element name="SparkThorProcess">
+        <xs:annotation>
+            <xs:documentation>Describes a spark cluster</xs:documentation>
```
Describes a Spark-Thor cluster

rpastrana commented on initfiles/componentfiles/configxml/sparkThor.xsd:
```diff
@@ -0,0 +1,232 @@
+            <!--DOC-Autobuild-code-->
+            <xs:annotation>
+                <xs:appinfo>
+                    <docid>Th.t1</docid>
```
@g-pan please review

rpastrana commented on initfiles/componentfiles/configxml/sparkThor.xsd:
```diff
@@ -0,0 +1,232 @@
+                </xs:appinfo>
+            </xs:annotation>  
+            <xs:sequence>
+                <xs:element name="Instance" maxOccurs="unbounded">
```
It's hard to visualize, but if this is a container of instances, it needs to be removed (sparkthor will inherit its instances from the referenced thor cluster)

rpastrana commented on initfiles/componentfiles/configxml/sparkThor.xsl.in:
```diff
@@ -0,0 +1,113 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+################################################################################
+#    HPCC SYSTEMS software Copyright (C) 2017 HPCC Systems®.
```
date

rpastrana commented on initfiles/componentfiles/configxml/sparkThor.xsl.in:
```diff
@@ -0,0 +1,113 @@
+        </xsl:call-template>
+        <xsl:text>SPARK_PID_DIR=@PID_PATH@</xsl:text>
+        <xsl:text>SPARK_CONF_DIR=@RUNTIME_PATH@/</xsl:text><xsl:value-of select="@name"/>
+    </xsl:template>
```
can we verify that NodeGroup/@name exists in the actual /Environment/Software/ThorCluster/@name ?

rpastrana commented on initfiles/sbin/spark-hpcc.sh.in:
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
```
this file might need renaming to sparkthor.sh.in

rpastrana commented on initfiles/sbin/spark-hpcc.sh.in:
```diff
@@ -0,0 +1,43 @@
+
+export SPARK_WORKER_DIR="@RUNTIME_PATH@/${_component}"
+if [[ ${_IS_THORSLAVE} == 0 ]]; then
+    @INSTALL_DIR@/externals/spark-hadoop/sbin/${_command}-slave.sh ${MASTER_URL}
```
is spark-hadoop a folder we create, or is it created by the spark install process?

Michael-Gardner replied:
It's a part of the spark install process.

rpastrana commented on initfiles/sbin/spark-hpcc.sh.in:
```diff
@@ -0,0 +1,43 @@
+_IS_THORSLAVE=$?
+# Always use the first record returned by `configgen -t esp` as master
+MASTER_IP="$(@ADMIN_PATH@/configgen -env @CONFIG_DIR@/@ENV_XML_FILE@ -listall -t esp | awk -F "," 'NR==1{print $3;}')"
+MASTER_URL="spark://${MASTER_IP}:7077"
```
is this port configurable?

rpastrana commented on spark/spark-defaults.conf.in:
```diff
@@ -0,0 +1,35 @@
+#
```
@fvillanustre as part of the spark-hpcc integration effort, we'll be supplying spark config files dynamically configured. However those files include the following license. Are we ok to do this?

rpastrana reviewed (changes requested):
@Michael-Gardner a few comments. We need to add a sparkthor to the stock env.xml (when use_spark) used to build/package. Also, I don't see where we pull information from the newly created config to dynamically configure spark.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: ea5f60bf5d9589c5e17f194622b30cd49dfa9a2f
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   19 sec (00:00:19) |  170 sec (00:02:50) |   93 sec (00:01:33) |   20 sec (00:00:20) |   42 sec (00:00:42) | 1348 sec (00:22:28) |   20 sec (00:00:20) | 1712 sec (00:28:32) |

rpastrana reviewed (changes requested):
@Michael-Gardner

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: fd9ff2919c128a92eadd97ed3dc9071319bbf12e
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   10 sec (00:00:10) |  176 sec (00:02:56) |   93 sec (00:01:33) |   19 sec (00:00:19) |   42 sec (00:00:42) | 1331 sec (00:22:11) |   20 sec (00:00:20) | 1691 sec (00:28:11) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: c9113d04a7fe832cbb51efdcde255be38022253a
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    8 sec (00:00:08) |  174 sec (00:02:54) |   93 sec (00:01:33) |   21 sec (00:00:21) |   43 sec (00:00:43) | 1356 sec (00:22:36) |   19 sec (00:00:19) | 1714 sec (00:28:34) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 3d3f5d681b78b0d14c66b670f3c6e79da75ed8e1
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   19 sec (00:00:19) |  172 sec (00:02:52) |   93 sec (00:01:33) |   20 sec (00:00:20) |   42 sec (00:00:42) | 1337 sec (00:22:17) |   20 sec (00:00:20) | 1703 sec (00:28:23) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: a742c76d77fc87f87d9aa58cc1e0d66794e17e09
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   15 sec (00:00:15) |  173 sec (00:02:53) |   94 sec (00:01:34) |   19 sec (00:00:19) |   41 sec (00:00:41) | 1333 sec (00:22:13) |   18 sec (00:00:18) | 1693 sec (00:28:13) |

rpastrana commented on initfiles/bin/init_sparkthor.in:
```diff
@@ -0,0 +1,57 @@
+logpthtail="$(date +%Y_%m_%d_%H_%M_%S)"
+export logfile="@LOG_PATH@/${component}/${component}_${logpthtail}.log"
+
+exec 3>&2 2>/var/log/HPCCSystems/${component}.debug
```
"/var/log/HPCCSystems" should be variable (@LOG_PATH@ perhaps) ?

rpastrana commented on initfiles/componentfiles/configxml/sparkThor.xsd:
```diff
@@ -0,0 +1,232 @@
+                    </xs:appinfo>
+                </xs:annotation>
+            </xs:attribute>
+            <xs:attribute name="SPARK_MASTER_HOST">
```
this should be a configmanager computer instance

rpastrana commented on initfiles/sbin/CMakeLists.txt:
```diff
@@ -37,6 +37,15 @@ ADD_CUSTOM_TARGET(ProcessFiles-initfiles-sbin ALL DEPENDS ${outFiles} ${noInstal
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/regex.awk.in.cmake" "${CMAKE_CURRENT_BINARY_DIR}/regex.awk")
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/backup.sh.in" "${CMAKE_CURRENT_BINARY_DIR}/backup.sh" @ONLY)
 
+if(USE_SPARK)
```
we also need to configure HPCC-Platform/initfiles/etc/DIR_NAME/environment.xml.in
 if USE_SPARK, to add stock "mysparkthor" associated with "mythor" and sparkmaster="myesp/Instance"

Michael-Gardner replied:
It is now included in the default environment.xml

rpastrana commented on initfiles/sbin/spark-hpcc-slave.sh.in:
```diff
@@ -0,0 +1,30 @@
+cd @RUNTIME_PATH@/${_component}
+
+MASTER_IP="$(@ADMIN_PATH@/configgen -env @CONFIG_DIR@/@ENV_XML_FILE@ -listall -t sparkthor | awk -F "," '{print $3;}')"
+MASTER_URL="spark://${MASTER_IP}:7077"
```
is the spark master port configurable?

rpastrana reviewed (changes requested):
@Michael-Gardner a few comments.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 5430e8658d9a07eaae2fa27ce3d3559220a9740a
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   10 sec (00:00:10) |  173 sec (00:02:53) |   94 sec (00:01:34) |   19 sec (00:00:19) |   41 sec (00:00:41) | 1347 sec (00:22:27) |   19 sec (00:00:19) | 1703 sec (00:28:23) |

Michael-Gardner commented:
*Note  -- use daliadmin instead of configgen to build the worker list

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 231343ceb04ef19a9cf5cdbc84820dba64e223e4
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   37 sec (00:00:37) |  174 sec (00:02:54) |   93 sec (00:01:33) |   19 sec (00:00:19) |   41 sec (00:00:41) | 1349 sec (00:22:29) |   19 sec (00:00:19) | 1732 sec (00:28:52) |

Michael-Gardner commented:
@rpastrana 
- included in the default environment.xml.
- now uses daliadmin to build the workers list
- renamed from spark-hpcc to sparkthor for all relevant scripts
- fixed issue of workers not starting from correct directory  on remote machines

rpastrana commented on initfiles/componentfiles/configxml/sparkThor.xsd:
```diff
@@ -108,52 +88,18 @@
                 </xs:annotation>
             </xs:attribute>
-            <xs:attribute name="SPARK_LOCAL_IP" type="xs:nonNegativeInteger">
+            <xs:attribute name="daliServers" type="daliServersType" use="required">
```
can this be inherited from the thor cluster config or from the overall environmen?

rpastrana commented on initfiles/componentfiles/configxml/sparkThor.xsd:
```diff
@@ -206,27 +145,13 @@
                 </xs:annotation>
             </xs:attribute>
-            <xs:attribute name="SPARK_WORKER_PORT" type="xs:nonNegativeInteger" default="7070">
+            <xs:attribute name="SPARK_WORKER_PORT" type="xs:nonNegativeInteger" default="7071">
```
this is b/c dali tends to run on 7070?

Michael-Gardner replied:
Yes.  And this gets written out to the spark-env.sh file so all the workers will be able to see the new default port.

rpastrana reviewed (commented):
@Michael-Gardner looks good, just a couple of comments

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 0cdc0817e213e2c6b4fa292c69a8efb6636431a5
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   22 sec (00:00:22) |  174 sec (00:02:54) |   95 sec (00:01:35) |   22 sec (00:00:22) |   44 sec (00:00:44) | 1519 sec (00:25:19) |   21 sec (00:00:21) | 1897 sec (00:31:37) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 316833021c85c5630afbcda0e8d48c88e870b2b7
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |806 |806 |0 |
|test (thor) |730 |730 |0 |
|test (roxie) |879 |879 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   11 sec (00:00:11) |  179 sec (00:02:59) |   95 sec (00:01:35) |   20 sec (00:00:20) |   42 sec (00:00:42) | 1366 sec (00:22:46) |   21 sec (00:00:21) | 1734 sec (00:28:54) |

Michael-Gardner commented:
@rpastrana Squashed the commit and kept the history of all our changes.

rpastrana reviewed (approved):
@Michael-Gardner 
Looks good.
@jpmcmu and @gfortil tested

Michael-Gardner commented:
@richardkchapman Ready for merge.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: b13a5eaad85ba13de35830118310566beb5d59a3
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |806 |806 |0 |
|test (thor) |730 |730 |0 |
|test (roxie) |879 |879 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   37 sec (00:00:37) |  185 sec (00:03:05) |   94 sec (00:01:34) |   20 sec (00:00:20) |   42 sec (00:00:42) | 1348 sec (00:22:28) |   18 sec (00:00:18) | 1744 sec (00:29:04) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 2430676dae6db665cebfe05d82ddf13aee1031b4
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |806 |806 |0 |
|test (thor) |730 |730 |0 |
|test (roxie) |879 |879 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   12 sec (00:00:12) |  189 sec (00:03:09) |   94 sec (00:01:34) |   21 sec (00:00:21) |   44 sec (00:00:44) | 1360 sec (00:22:40) |   19 sec (00:00:19) | 1739 sec (00:28:59) |
